pkg/router/template/router_test.go

```diff
@@ -991,20 +1097,18 @@ func TestCalculateServiceWeights(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		for suKey, eps := range tc.serviceUnits {
-			router.CreateServiceUnit(suKey)
-			router.AddEndpoints(suKey, eps)
-		}
-		endpointWeights := router.calculateServiceWeights(tc.serviceWeights)
-		if !reflect.DeepEqual(endpointWeights, tc.expectedWeights) {
-			t.Errorf("test %s: expected endpointWeights to be %v, got %v", tc.name, tc.expectedWeights, endpointWeights)
-		}
-		// Remove endpoints and service units so the same sample template router
-		// can be re-used.
-		for suKey := range tc.serviceUnits {
-			router.DeleteEndpoints(suKey)
-			router.DeleteServiceUnit(suKey)
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			router := NewFakeTemplateRouter()
+
+			for suKey, eps := range tc.serviceUnits {
+				router.CreateServiceUnit(suKey)
+				router.AddEndpoints(suKey, eps)
+			}
+			endpointWeights := router.calculateServiceWeights(tc.serviceWeights, tc.routePort)
+			if !reflect.DeepEqual(endpointWeights, tc.expectedWeights) {
+				t.Errorf("test %s: expected endpointWeights to be %v, got %v", tc.name, tc.expectedWeights, endpointWeights)
```

Comment:
You don't need to include the test name in the error message; using `t.Run` gets you that for free.

Reply:
Done